src/Raven.Client/Http/RavenCommand.cs

```diff
@@ -84,7 +84,7 @@ public bool IsFailedWithNode(ServerNode node)
         {
             using (response)
             {
-                if (ResponseType == RavenCommandResponseType.Empty)
+                if (ResponseType == RavenCommandResponseType.Empty || response.StatusCode == HttpStatusCode.NoContent)
```

Comment:
Actually, that class is part of the test infrastructure, meaning we could just add another class, there is another issue though: I think that the RavenCommand class should have an overload that receives a response type